client/src/pages/lessons/edit.tsx

```diff
@@ -425,467 +615,539 @@ const LessonEdit = (props: {
+                          startLessonTraining(lessonUnderEdit.lesson);
+                        }
+                      }}
+                    >
+                      {drawerOpen ? "TRAIN AI" : ""}
+                    </Button>
+                  </ListItem>
+                  <ListItem
+                    sx={{
+                      marginTop: 0,
+                      paddingTop: 0,
+                      display: drawerOpen ? "flex" : "none",
+                    }}
+                  >
+                    <Grid
+                      container
+                      direction="column"
+                      alignItems="center"
+                      spacing={1}
+                    >
+                      <Grid item>
+                        <Typography variant="caption">
+                          {`Last Trained: ${lastTrainedString}`}
+                        </Typography>
+                      </Grid>
+                      <Divider />
+                      <Grid item>
+                        <ListItem>
+                          {isTraining ? (
+                            <LoadingIndicator />
+                          ) : trainStatus.state === TrainState.SUCCESS ? (
+                            <List>
+                              {trainStatus.info?.expectations?.map((x, i) => (
+                                <ListItem key={`train-success-accuracy-${i}`}>
+                                  <ListItemText
+                                    style={{ textAlign: "center" }}
+                                    data-cy={`train-success-accuracy-${i}`}
+                                  >{`Expectation ${
+                                    i + 1
+                                  } Accuracy: ${x.accuracy.toFixed(
+                                    2
+                                  )}`}</ListItemText>
+                                </ListItem>
+                              ))}
+                            </List>
+                          ) : trainStatus.state === TrainState.FAILURE ? (
+                            <Typography data-cy="train-failure">{`Training Failed`}</Typography>
+                          ) : null}
+                        </ListItem>
+                      </Grid>
+                    </Grid>
+                  </ListItem>
+                </ThemeProvider>
+                <ListItem>
+                  {isDownloadable ? (
+                    <Button
+                      data-cy="download-button"
+                      variant="contained"
+                      startIcon={<Download />}
+                      color="primary"
+                      size="large"
+                      onClick={download}
+                      disabled={isDownloading}
+                    >
+                      Download
+                    </Button>
+                  ) : null}
+                </ListItem>
+              </List>
+            </div>
+          </Drawer>
+        </Grid>
```

Comment:
something about this smells, It feels like you're trying to do too much in the view portion of the codebase.  If this really is all in UI code, then consider breaking it down into smaller components